src/main/java/com/khalsa_ji/ems/rest/Post.java

```diff
@@ -0,0 +1,137 @@
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Locale;
+
+/**
+ * The class {@code Post} is a <em>REST Controller</em> class, which aims to listen to the
+ * <strong>POST</strong> requests made at the <strong>"/api/v1"</strong> endpoint.
+ *
+ * @author Ravikiran Singh
+ * @version 1.0
+ * @since 1.0
+ * @see Employee
+ */
+
+@RestController
+@RequestMapping(value = "/api/v1")
+public class Post {
+    @Autowired
+    EmployeeService employeeService;
+
+    @Autowired
+    DesignationService designationService;
+
+    @Autowired
+    MessageSource messageSource;
+
+    private final String[] EMPLOYEE_NAME = {"Employee name"};
+    private final String[] JOB_TITLE = {"Job Title(Designation)"};
+    private final String[] MANAGER = {"Manager"};
+
+    /**
+     * Method to register(add) a new instance of the {@code Employee} class
+     *
+     * @param employee An instance of the {@code Employee} class
+     * @return A saved instance of the {@code Employee} class
+     * @throws URISyntaxException URI Syntax Exception
+     * @see Employee
+     */
+
+    @PostMapping(value = "/employees", consumes = "application/json", produces = "application/json")
+    @ApiOperation(value = "Adds a new Employee", response = Employee.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Employee added successfully"),
+            @ApiResponse(code = 400, message = "Invalid details found for adding a new employee")
+    })
+    public ResponseEntity<Object> addEmployee(@RequestBody Employee employee) throws URISyntaxException {
+        //  Validating employee's name
+        switch(Validator.validateString(employee.getEmployeeName())) {
+            case nullString:        return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.nullString", EMPLOYEE_NAME, Locale.getDefault()));
+            case emptyString:       return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.emptyString", EMPLOYEE_NAME, Locale.getDefault()));
+            case invalidString:     return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.invalidString", EMPLOYEE_NAME, Locale.getDefault()));
+        }
```

Comment:
These can be converted to some generic validation method, which will be used at various places.

Reply:
Agreed, would implement the same.